source/wp-content/themes/wporg-themes-2024/src/theme-downloads/index.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Block Name: Downloads Chart
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\Theme\Theme_Directory_2024\Theme_Downloads_Block;
+
+defined( 'WPINC' ) || die();
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Register the block.
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/theme-downloads',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+
+	// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion -- third-party script.
+	wp_register_script( 'google-charts-loader', 'https://www.gstatic.com/charts/loader.js', array(), null, true );
+}
+
+/**
+ * Render the block content.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	if ( ! $block->context['postId'] ) {
+		return '';
+	}
+
+	$theme_post = get_post( $block->context['postId'] );
+	$theme = wporg_themes_theme_information( $theme_post->post_name );
```

Comment:
You may want to check this return value more specifically in case it returns an [error](https://github.com/WordPress/wordpress.org/blob/ed2884a3cfbfb45bf0ba327c67643d96ca5bfeeb/wordpress.org/public_html/wp-content/plugins/theme-directory/class-themes-api.php#L502).

Reply:
Done in https://github.com/WordPress/wporg-theme-directory/pull/54/commits/e005f29c1c1362d2f9ef40901144519d86d9d689, also catching any errors from the API in https://github.com/WordPress/wporg-theme-directory/pull/54/commits/5997c9f3cf814ff8e410a8baf5fe4bcd5eda722b.